src/main/rx/expresslrs.c

```diff
@@ -1059,65 +1057,134 @@ static void enterBindingMode(void)
+        }
 
-    if (rxSpiCheckBindRequested(true)) {
-        enterBindingMode();
-    }
+        if (rxSpiCheckBindRequested(true)) {
+            enterBindingMode();
+        }
 
-    uint8_t irqReason = receiver.rxISR(&isrTimeStampUs);
-    if (irqReason == ELRS_DIO_TX_DONE) {
-        startReceiving();
-    } else if (irqReason == ELRS_DIO_RX_DONE) {
-        result = processRFPacket(payload, isrTimeStampUs);
-    }
+        irqReason = receiver.rxISR(&isrTimeStampUs);
+
+#ifdef USE_ELRS_STATE_ISR
+        expressLrsState = RX_ELRS_STATE_ISR;
+        break;
+
+    case RX_ELRS_STATE_ISR:
+#endif
+        expressLrsState = RX_ELRS_STATE_FHSS;
 
-    if (receiver.fhssRequired) {
-        receiver.fhssRequired = false;
-        bool didFHSS = false;
-        bool tlmReq = false;
-        ATOMIC_BLOCK(NVIC_PRIO_TIMER) { // prevent from updating nonce in TICK
-            didFHSS = handleFHSS();
-            tlmReq = shouldSendTelemetryResponse();
+        if (irqReason == ELRS_DIO_TX_DONE) {
+            startReceiving();
+        } else if (irqReason == ELRS_DIO_RX_DONE) {
+            receiver.receiveData(packet, ELRS_RX_TX_BUFF_SIZE);
+
+            expressLrsState = RX_ELRS_STATE_PROCESS;
         }
 
-        if (tlmReq) {
-            // in case we miss a packet before TLM we still need to estimate processing time using %
-            uint32_t processingTime = (micros() - isrTimeStampUs) % receiver.modParams->interval;
-            if (processingTime < PACKET_HANDLING_TO_TOCK_ISR_DELAY_US && receiver.timerState == ELRS_TIM_LOCKED) {
-                handleSendTelemetryResponse();
-            } else {
-                receiver.alreadyTLMresp = true;
-                startReceiving();
-            }
+#ifdef USE_ELRS_STATE_PROCESS
+        break;
+    case RX_ELRS_STATE_PROCESS:
+#else
+        if (expressLrsState == RX_ELRS_STATE_PROCESS)
+#endif
+        {
+            pinioSet(2, 1);
+            result = processRFPacket(payload, isrTimeStampUs);
+            pinioSet(2, 0);
         }
 
-        if (rxExpressLrsSpiConfig()->domain != ISM2400 && !didFHSS && !tlmReq && lqPeriodIsSet()) {
-            receiver.handleFreqCorrection(receiver.freqOffset, receiver.currentFreq); //corrects for RX freq offset
+#ifdef USE_ELRS_STATE_FHSS
+        expressLrsState = RX_ELRS_STATE_FHSS;
+        break;
+
+    case RX_ELRS_STATE_FHSS:
+#endif
+          if (receiver.fhssRequired) {
+            receiver.fhssRequired = false;
+            bool didFHSS = false;
+            bool tlmReq = false;
+            ATOMIC_BLOCK(NVIC_PRIO_TIMER) { // prevent from updating nonce in TICK
+                didFHSS = handleFHSS();
+                tlmReq = shouldSendTelemetryResponse();
+            }
+
+            if (tlmReq) {
+                // in case we miss a packet before TLM we still need to estimate processing time using %
+                uint32_t processingTime = (micros() - isrTimeStampUs) % receiver.modParams->interval;
```

Comment:
possibly `cmpTimeUs`, but negative delta time has to be handled

Reply:
Changed to:
```
deltaTimeUs processingTime = cmpTimeUs(micros(), isrTimeStampUs) % receiver.modParams->interval;
```